src/azure-cli/azure/cli/command_modules/iot/commands.py

```diff
@@ -59,13 +59,17 @@ def load_command_table(self, _):  # pylint: disable=too-many-statements
         g.generic_update_command('update', getter_name='iot_dps_get', setter_name='iot_dps_update',
                                  command_type=update_custom_util)
 
-    # iot dps access-policy commands
-    with self.command_group('iot dps access-policy', client_factory=iot_service_provisioning_factory) as g:
-        g.custom_command('list', 'iot_dps_access_policy_list')
-        g.custom_show_command('show', 'iot_dps_access_policy_get')
-        g.custom_command('create', 'iot_dps_access_policy_create', supports_no_wait=True)
-        g.custom_command('update', 'iot_dps_access_policy_update', supports_no_wait=True)
-        g.custom_command('delete', 'iot_dps_access_policy_delete', supports_no_wait=True)
+    # iot dps access-policy commands (Deprecated)
+    with self.command_group('iot dps access-policy',
+                            client_factory=iot_service_provisioning_factory,
+                            deprecate_info=self.deprecate(redirect='iot dps policy',
+                                                          expiration='2.35.0')
+                            ) as g:
+        g.custom_command('list', 'iot_dps_policy_list')
+        g.custom_show_command('show', 'iot_dps_policy_get')
+        g.custom_command('create', 'iot_dps_policy_create', supports_no_wait=True)
+        g.custom_command('update', 'iot_dps_policy_update', supports_no_wait=True)
+        g.custom_command('delete', 'iot_dps_policy_delete', supports_no_wait=True)
```

Comment:
Do we need to consider adding `confirmation=True` for the delete commands?

Reply:
While I believe we can add these for the [new delete commands](https://github.com/Azure/azure-cli/pull/20682/files/44f8e578e287538bbb49fe1a2fd8a1d1a0c38896#diff-e02408e4f1ccd9dbaf9422c71a7e68d0b1f119e49c8e4713de91c47b2f33ec75R93-R99), I believe this would then be a breaking change for users using the old commands?

If we add `confirmation=True` to the existing (deprecated) commands, users with existing workflows may break waiting for confirmation on a command that didn't require confirmation before.

Reply:
OK